importer.py

```diff
@@ -235,18 +278,27 @@ def print_not_imported(self):
 
     parser.add_argument('-S', '--strict-artists-search', help='Search for an exact match of all artists', default=False)
 
+    parser.add_argument('-j', '--json', help='JSON file to import tracks from')
+
     arguments = parser.parse_args()
 
-    auth_manager = SpotifyOAuth(
-        client_id=arguments.id,
-        client_secret=arguments.secret,
-        redirect_uri=REDIRECT_URI,
-        scope='playlist-modify-public, user-library-modify, user-follow-modify, ugc-image-upload',
-        username=arguments.spotify,
-        cache_path='cache.txt'
-    )
-    spotify_client_ = spotipy.Spotify(auth_manager=auth_manager, requests_timeout=arguments.timeout)
-
-    yandex_client_ = Client(arguments.token)
-    yandex_client_.init()
-    Importer(spotify_client_, yandex_client_, arguments.ignore, arguments.strict_artists_search).import_all()
+    try:
+        auth_manager = SpotifyOAuth(
+            client_id=arguments.id,
+            client_secret=arguments.secret,
+            redirect_uri=REDIRECT_URI,
+            scope='playlist-modify-public, user-library-modify, user-follow-modify, ugc-image-upload',
+            username=arguments.spotify,
+            cache_path='cache.txt'
+        )
+        spotify_client_ = spotipy.Spotify(auth_manager=auth_manager, requests_timeout=arguments.timeout)
+
+        yandex_client_ = Client(arguments.token)
```

Comment:
token looks required as before

Reply:
as said before, this change allows users to use JSON only if needed